### PR TITLE
Feature/news: US10 add my posted news in news bloc

### PR DESCRIPTION
**Pull request by aelbenney**

### CODE REVIEW

File: services/src/main/java/org/exoplatform/news/queryBuilder/NewsQueryBuilder.java [hunk 1]
```diff
@@ -33,6 +33,9 @@ public StringBuilder buildQuery(NewsFilter filter) throws Exception{
             }
             sqlQuery.append("exo:spaceId = '").append(spaces.get(spaces.size()-1)).append("') AND ");
           }
+          if (filter.getAuthor() != null && !filter.getAuthor().equals("")) {
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/queryBuilder/NewsQueryBuilder.java]:
```suggestion
          if (StringUtils.isNotEmpty(filter.getAuthor())) {
```

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 2]
```diff
@@ -184,11 +184,17 @@ public Response getNews(@Context HttpServletRequest request,
...
-  private NewsFilter buildFilter(List<String> spaces, String filter, String text) {
+  private NewsFilter buildFilter(List<String> spaces, String filter, String text, String author) {
     NewsFilter newsFilter = new NewsFilter();
     newsFilter.setSpaces(spaces);
-    if (("pinned").equals(filter) && StringUtils.isNotEmpty(filter)) {
-      newsFilter.setPinnedNews(true);
+    if (StringUtils.isNotEmpty(filter)) {
+      if ("pinned".equals(filter)) {
+        newsFilter.setPinnedNews(true);
+      } else if ("myPosted".equals(filter)) {
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: We should create a enum for the filter value IMO. WDYT ?

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 3]
```diff
@@ -184,11 +184,17 @@ public Response getNews(@Context HttpServletRequest request,
...
+        if (StringUtils.isNotEmpty(author)) {
+          newsFilter.setAuthor(author);
+        }
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: Can we add unit test for this case ?

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java (hunk 2 above)
Comment (aelbenney) [reply services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: Yes, right (y) I will create it

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java (hunk 3 above)
Comment (aelbenney) [reply services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]: okay

File: services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java [hunk 4]
```diff
@@ -184,11 +188,24 @@ public Response getNews(@Context HttpServletRequest request,
...
   }
 
-  private NewsFilter buildFilter(List<String> spaces, String filter, String text) {
+  private NewsFilter buildFilter(List<String> spaces, String filter, String text, String author) {
     NewsFilter newsFilter = new NewsFilter();
     newsFilter.setSpaces(spaces);
-    if (("pinned").equals(filter) && StringUtils.isNotEmpty(filter)) {
-      newsFilter.setPinnedNews(true);
+    if (StringUtils.isNotEmpty(filter)) {
+      FilterType filterType=  FilterType.valueOf(filter.toUpperCase());
```
Comment (thomasdelhomenie) [inline services/src/main/java/org/exoplatform/news/rest/NewsRestResourcesV1.java]:
```suggestion
      FilterType filterType = FilterType.valueOf(filter.toUpperCase());
```

**Pull request merged**